module/documents/item.mjs

```diff
@@ -606,6 +606,32 @@ export default class Item5e extends SystemDocumentMixin(Item) {
       }, []);
     }
 
+    // Price
+    if ( ("price" in this.system) && ("gp" in CONFIG.DND5E.currencies) ) {
+      const { value, denomination } = this.system.price;
+      const { conversion } = CONFIG.DND5E.currencies[denomination] ?? {};
+      const { gp } = CONFIG.DND5E.currencies;
+      if ( conversion ) {
+        const multiplier = gp.conversion / conversion;
+        this.system.price.valueInGP = Math.floor(value * multiplier);
+      }
+    }
+
+    // Type
+    if ( "type" in this.system ) {
+      let identifier = "";
+      const { baseItem, value } = this.system.type;
+      switch ( this.type ) {
+        case "weapon": identifier = CONFIG.DND5E.weaponIds[baseItem]; break;
+        case "tool": identifier = CONFIG.DND5E.toolIds[baseItem]; break;
+        case "equipment":
+          if ( value === "shield" ) identifier = CONFIG.DND5E.shieldIds[baseItem];
+          else identifier = CONFIG.DND5E.armorIds[baseItem];
+          break;
+      }
+      this.system.type.identifier = identifier;
+    }
```

Comment:
Maybe these should move to their associated templates? But then we need to make sure every item type is calling these template preparation methods manually, and I wasn't sure if that was the convention we decided on.

Reply:
That is generally the pattern I've been adopting, like we have `prepareDescriptionData` for `ItemDescriptionTemplate` already. We just have to make sure things don't immediately break if a module sub-type doesn't call them.

Reply:
Moved price preparation to `PhysicalItemTemplate` and hardened code paths to not assume `price.valueInGP` exists. I moved the `type.identifier` preparation to `WeaponData`, `ToolData`, etc. since it is dependent on the actual Item type so it made sense to just do the preparation in that exact context.